specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json

```diff
@@ -1248,7 +1259,12 @@
         "authenticationType": {
           "type": "object",
           "x-ms-format": "dfe-string",
-          "description": "The authentication type to connect to Common Data Service for Apps server. 'Office365' for online scenario, 'Ifd' for on-premises with Ifd scenario. 'AADServicePrincipal' for Server-To-Server authentication in online scenario. Type: string (or Expression with resultType string)."
+          "description": "The authentication type to connect to Common Data Service for Apps server. 'Office365' for online scenario, 'Ifd' for on-premises with Ifd scenario. 'AADServicePrincipal' for Server-To-Server authentication in online scenario, 'Active Directory' for Dynamics on-premises with IFD. Type: string (or Expression with resultType string)."
+        },
+        "domain": {
+          "type": "object",
+          "x-ms-format": "dfe-string",
```

Comment:
why have you defined all these properties as objects ? Shouldnt these be defined as strings ?

Reply:
ADF support expression for every property, so we need object instead of string.
https://learn.microsoft.com/en-us/azure/data-factory/control-flow-expression-language-functions

Reply:
I dont understand. The document did not make this clear. 

In ARM it is not a best practice to have freeform objects in the property bag. I am surprised you have not been flagged for this before. Is this something you have already discussed with the ARM team before as an exception ?

Reply:
Yes, ADF have this exception.
In this link, for example
"name": "value" is a value of string
"name": "@pipeline().parameters.password"  is a value of object, we will revolve this value from expression and result will be a string.